src/main/java/core/basesyntax/Lottery.java

```diff
@@ -0,0 +1,16 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class Lottery extends ColorSupplier {
```

Comment:
There is no need for `Lottery` to extend `ColorSupplier` class. Lottery has to use ColorSupplier to generate a proper `Ball` instance, and therefore `ColorSupplier` should be a private field of  `Lottery` class.